File: idling-registry-okhttp/app/src/debug/java/com/sqisland/espresso/idling_registry/okhttp/IdlingResources.java
package com.sqisland.espresso.idling_registry.okhttp;

import android.support.test.espresso.IdlingRegistry;

import com.jakewharton.espresso.OkHttp3IdlingResource;

import okhttp3.OkHttpClient;

public abstract class IdlingResources {
  public static void registerOkHttp(OkHttpClient client) {
    IdlingRegistry.getInstance().register(OkHttp3IdlingResource.create(
        "okhttp", client));
  }
}

File: idling-registry-okhttp/app/src/debug/java/com/sqisland/espresso/idling_registry/okhttp/IdlingResources.kt
package com.sqisland.espresso.idling_registry.okhttp

import android.support.test.espresso.IdlingRegistry

import com.jakewharton.espresso.OkHttp3IdlingResource

import okhttp3.OkHttpClient

object IdlingResources {
  fun registerOkHttp(client: OkHttpClient) {
    IdlingRegistry.getInstance().register(OkHttp3IdlingResource.create(
      "okhttp", client))
  }
}